app/status.go

```diff
@@ -363,3 +364,86 @@ func (a *App) GetStatus(userID string) (*model.Status, *model.AppError) {
 func (a *App) IsUserAway(lastActivityAt int64) bool {
 	return model.GetMillis()-lastActivityAt >= *a.Config().TeamSettings.UserStatusAwayTimeout*1000
 }
+
+func (a *App) SetCustomStatus(userID string, cs *model.CustomStatus) *model.AppError {
+	user, err := a.GetUser(userID)
+	if err != nil {
+		return err
+	}
+
+	user.SetCustomStatus(cs)
+	_, updateErr := a.UpdateUser(user, true)
+	if updateErr != nil {
+		return err
+	}
+
+	if err := a.addRecentCustomStatus(userID, cs); err != nil {
+		return err
```

Comment:
I'm not 100% sure if we should fail entirely if you are not able to add the custom status to the recent custom statuses. Maybe we can log that. The custom status is already changed, so a "partial" update is better than a confusing response.

Reply:
@jespino Agree that it may be a bit confusing to return an error as the `API` or `Slash command` response even after the status is already changed in lieu of failed recent statuses update.

As per your suggestion, I have updated to only log the error in this case.